### PR TITLE
GH-292: SADL Web

### LINKED ISSUE #292: SADL Web
This task is the umbrella task for the Web SADL.

The development is done on the [`sadl_theia`](https://github.com/crapo/sadlos2/tree/sadl_theia) branch.

### PR DESCRIPTION
**Pull request by kittaakos**
The followings are the significant Java code changes:

 - [`IPreferenceValuesProvider` provider for the Web SADL](https://github.com/crapo/sadlos2/pull/296/files#diff-58e93c54b81e3bc7cc6d50157296a4f9R37): To be able to use the project-specific `settings.json`.
 - [`SADLValidator` extension for the Web SADL](https://github.com/crapo/sadlos2/pull/296/files#diff-71fedb9ccb73b4fbed8a5c5568b1e3e4R37): To bail validation if a model is not in a SADL project. I added that to avoid unnecessary model processor and model validator code changes.
 - [The language server extension for SADL is a SADL console](https://github.com/crapo/sadlos2/pull/296/files#diff-d402c1155211eef53788359d8323d73dR45): It is for sending back messages to the `Output` view in Theia.
 - [Customized workspace manager](https://github.com/crapo/sadlos2/pull/296/files#diff-141a473f8540bad13875b9e9a9fffba9R30) to avoid CCE when working on External EMF resources.
 - [Customized incremental builder](https://github.com/crapo/sadlos2/pull/296/files#diff-b7bae2ee9a1f6f42edddbe50a0ef17c6R34) to be able to support multiple SADL projects per workspace.

Closes #292.

### CODE REVIEW

**Comment by kittaakos** [comment]
I have fixed the conflict, changed the base branch from the `master` to `development` and squashed the commits so that it can be consumed via humans too.

**Comment by kittaakos** [comment]
@crapo, is there any chance get this PR merged? I am trying to resolve the conflicts from time to time, but it is getting hard to maintain this branch. Thanks!

**Comment by crapo** [comment]
I got pulled away but now I'm back on task. I think I should be able to build the Web Sadl docker image and complete initial understanding of how it works as I evaluate the merging of this branch. We just need to make sure that we don't break development. I'll create a branch off development for merge and testing.

**Comment by svenefftinge** [comment]
@kittaakos maybe you should rebase again

**Comment by kittaakos** [comment]
TODOs:
 -  [ ] resolve the merge conflicts.
 -  [ ] rebase from the `development` branch.
 -  [ ] build a new language server for SADL.
 -  [ ] publish a new Theia extension for SADL into `npmjs.org`.
 -  [ ] build and publish a new Docker image for SADL.
 -  [ ] verify it on Windows too.

Not a must but would be great to have:
 -  [ ] bump up to latest Xtext (2.15).
 -  [ ] bump up to latest Theia.

**Comment by crapo** [comment]
BTW, the process

1.  build a new language server for SADL.
2.  publish a new Theia extension for SADL into npmjs.org.
3.  build and publish a new Docker image for SADL.

is one of my questions. How to do that wasn't obvious to me from the documentation. If you could make some more notes as you do this it would be very helpful.

**Comment by kittaakos** [comment]
> If you could make some more notes as you do this it would be very helpful.

It is a part of the extension (from July): https://www.npmjs.com/package/sadl-extension#release-engineering/

We could make it happen on the CIs, but 
 - I do not know if I can configure it due to the lack of admin rights,
 - no one works on PRs in the repository.

Maybe we can have a CRON job, that runs each night.

File: sadl3/com.ge.research.sadl.parent/com.ge.research.sadl.ide/src/com/ge/research/sadl/docgen/GenerateKeywordSymbolHtmlTable.java [hunk 1]
```diff
@@ -1,110 +0,0 @@
...
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
-
-import com.ge.research.sadl.ide.contentassist.antlr.internal.InternalSADLParser;
-
-
-
-public class GenerateKeywordSymbolHtmlTable {
```
Comment (kittaakos) [inline sadl3/com.ge.research.sadl.parent/com.ge.research.sadl.ide/src/com/ge/research/sadl/docgen/GenerateKeywordSymbolHtmlTable.java]: TODO: revert the deletion.

**Comment by crapo** [comment]
Merge with development branch completed.

**Pull request merged**